db/migration/1719842654592-AddChartConfigsTable.ts

```diff
@@ -0,0 +1,134 @@
+import { MigrationInterface, QueryRunner } from "typeorm"
+
+export class AddChartConfigsTable1719842654592 implements MigrationInterface {
+    private async createChartConfigsTable(
+        queryRunner: QueryRunner
+    ): Promise<void> {
+        await queryRunner.query(`-- sql
+            CREATE TABLE chart_configs (
+                id binary(16) NOT NULL DEFAULT (UUID_TO_BIN(UUID(), 1)) PRIMARY KEY,
+                uuid varchar(36) GENERATED ALWAYS AS (BIN_TO_UUID(id, 1)) VIRTUAL,
+                patch json NOT NULL,
+                full json NOT NULL,
```

Comment:
@danyx23 I initially thought it would be handy to always have the "full" config at hand, so I made the field non-nullable. And I think that still holds true for proper grapher charts, but for indicator charts, there isn't a real upside to having the "full" config precomputed, but it is quite cumbersome to make sure the full config gets updated whenever the patch is updated. I think I'd like to allow nulls for the full config, and then leave it empty for indicator charts. What do you think?

Just food for thought: If we went one step further, we could think about `chart_configs` as a table of patches, remove the full config column, and instead add back a `config` column to the charts table that holds the full config. Not sure this is a good idea but that way we don't have to do a ton of joins and we can make the config column on the charts table non-nullable.

Reply:
I just asked this question in our meeting. As a recap:
- I'll make `chart_configs.full` nullable and don't precompute full configs for indicator charts
- It's an open question where the full config of our grapher/stand-alone charts should live then, in `chart_configs.full` or `charts.config`?

Reply:
I decided to take the original approach (the full config is non-nullable, always precomputed, and lives on the chart_configs table).